src/watertap_contrib/reflo/unit_models/multi_effect_crystallizer.py

```diff
@@ -0,0 +1,335 @@
+                tmp_dict["parameters"] = self.config.property_package_vapor
+                tmp_dict["defined_state"] = False
+
+                effect.heating_steam = (
+                    self.config.property_package_vapor.state_block_class(
+                        self.flowsheet().config.time,
+                        doc="Material properties of inlet heating steam",
+                        **tmp_dict,
+                    )
+                )
+
+                @effect.Constraint(
+                    doc="Change in temperature at inlet for first effect"
+                )
+                def eq_delta_temperature_inlet_effect_1(b):
+                    return (
+                        b.delta_temperature_in[0]
+                        == b.heating_steam[0].temperature - b.temperature_operating
+                    )
+
+                @effect.Constraint(
+                    doc="Change in temperature at outlet for first effect"
+                )
+                def eq_delta_temperature_outlet_effect_1(b):
+                    return (
+                        b.delta_temperature_out[0]
+                        == b.heating_steam[0].temperature
+                        - b.properties_in[0].temperature
+                    )
+
+                @effect.Constraint(doc="Heat transfer equation for first effect")
+                def eq_heat_transfer_effect_1(b):
+                    # return b.work_mechanical[0] == pyunits.convert(
+                    #     b.overall_heat_transfer_coefficient
+                    #     * b.heat_exchanger_area
+                    #     * b.delta_temperature[0]
+                    #     # , to_units=pyunits.kilowatt
+                    # )
+                    return (
+                        b.work_mechanical[0]
+                        == b.overall_heat_transfer_coefficient
+                        * b.heat_exchanger_area
+                        * b.delta_temperature[0]
+                    )
+
+                @effect.Constraint(doc="Calculate mass flow rate of heating steam")
+                def eq_heating_steam_flow_rate(b):
+                    return b.work_mechanical[0] == (
+                        pyunits.convert(
+                            b.heating_steam[0].dh_vap_mass
+                            * b.heating_steam[0].flow_mass_phase_comp["Vap", "H2O"],
+                            to_units=pyunits.kJ / pyunits.s,
+                        )
+                    )
+
+                self.add_port(name="inlet", block=effect.properties_in)
+                self.add_port(name="outlet", block=effect.properties_out)
+                self.add_port(name="solids", block=effect.properties_solids)
+                self.add_port(name="vapor", block=effect.properties_vapor)
+                self.add_port(name="pure_water", block=effect.properties_pure_water)
+                self.add_port(name="steam", block=effect.heating_steam)
+                self.steam.temperature.setub(1000)
```

Comment:
Not sure I understand the logic---this looks like you add ports for only the first effect, when I would imagine the outlets being specified for the last effect. Whatever the case, can you explain the logic here?

Reply:
Hm. This has made me think, I will have to revisit.

Reply:
Ok so this comment prompted to revisit the model and realized the inlet flows the user would set would not end up being the resulting inlet flows from the model. This would obviously be problematic if this model was connected to an upstream process.

This was addressed starting in 651dba5.

I added a ```ControlVolume0D``` to ensure a mass balance across the unit. The model initialization will now automatically deactivate the constraint that links energy flows between effects and initialize the water/salt mass flow rate in each effect with ```control_volume_mass_flow / number_effects```  (regardless of if the user fixed the mass flows for individual effects in the build) as if it was a standalone model.

Then, inlet flows to each effects are unfixed, linking constraints are re-activated, and the salt _concentration_ at the inlet for each effect is fixed to be the same across all effects. When the model is solved, the ```ControlVolume0D``` handles how much flow to send to each effect. You can see different approaches to take for the build and the resulting DOF in the ```test_dof``` tests.

Reply:
Oh I guess I never actually answered your question lol.

I moved the ```inlet``` and ```outlet``` to be on the control volume. As for why the rest are on the first effect, I guess I figured that is where they might be in an actual crystallizer? I don't really have a great reason. 

I actually think my moving to the control volume approach makes the ```pure_water``` port moot. I shall remove it.